include/fsm/fill_spill_merge.hpp

```diff
@@ -539,18 +546,19 @@ static dh_label_t OverflowInto(
   //SECOND PLACE TO STASH WATER: IN THIS DEPRESSION'S NEIGHBOUR
   //Maybe we can fit it into this depression's overflow depression!
 
+
   auto &pdep = deps.at(this_dep.parent);
   if(this_dep.odep==NO_VALUE){      //Does the depression even have such a neighbour? 
-    if(this_dep.parent!=OCEAN && pdep.water_vol==0) //At this point we're full and heading to our parent, so it needs to know that it contains our water
-      pdep.water_vol += this_dep.water_vol;
+    if(this_dep.parent!=OCEAN && pdep.water_vol==0 && (pdep.lchild == this_dep.dep_label || pdep.rchild == this_dep.dep_label)) //At this point we're full and heading to our parent, so it needs to know that it contains our water
+      pdep.water_vol += deps.at(pdep.lchild).dep_vol + deps.at(pdep.rchild).dep_vol;//this_dep.water_vol;
     return jump_table[root] = OverflowInto(this_dep.parent, stop_node, deps, jump_table, extra_water);  //Nope. Pass the water to the parent
   }
 
   //Can overflow depression hold more water?
   auto &odep = deps.at(this_dep.odep);
   if(odep.water_vol<odep.dep_vol){  //Yes. Move the water geographically into that depression's leaf.
-    if(this_dep.parent!=OCEAN && pdep.water_vol==0 && odep.water_vol+extra_water>odep.dep_vol) //It might take a while, but our neighbour will overflow, so our parent needs to know about our water volumes
-      pdep.water_vol += this_dep.water_vol + odep.dep_vol;           //Neighbour's water_vol will equal its dep_vol
+    if(this_dep.parent!=OCEAN && pdep.water_vol==0 && odep.water_vol+extra_water>odep.dep_vol && (pdep.lchild == odep.dep_label || pdep.rchild == odep.dep_label) && (pdep.lchild == this_dep.dep_label || pdep.rchild == this_dep.dep_label)) //It might take a while, but our neighbour will overflow, so our parent needs to know about our water volumes
+      pdep.water_vol += this_dep.dep_vol + odep.dep_vol;           //Neighbour's water_vol will equal its dep_vol
```

Comment:
Why?

Reply:
Again because I was concerned that water_vol may not == dep_vol. If we are passing water volume to our parent, it should be because we are full, so dep_vol is guaranteed to be the right amount of water to pass in.